### PR TITLE
Applying new conventions

### PR DESCRIPTION
**Pull request by zclimber**
Convention applying per #2 more or less done, althrough there's a couple of issues with it:
Several questions about modules have arisen.
1. Should classes/types be annotated like packages, like functions and operators or in some other way?
2. Same about modules: should they be annotated like packages, like functions and operators or in some another way?
3. When reference in text can be resolved to module or to class/type, for example, [Identity](http://hackage.haskell.org/package/base-4.10.1.0/docs/Data-Functor-Identity.html#t:Identity) module and [_`Identity`_](http://hackage.haskell.org/package/base-4.10.1.0/docs/Data-Functor-Identity.html#t:Identity) type, what variant should be chosen?
4. Didn't find MonadIO and CoroutineT from Lecture 9, Stream comonad from Lecture 13.
5. Unclear how to format Lecture 16: Idris and Uncovered Topics (left them as is).

### CODE REVIEW

**Comment by chshersh** [comment]
@zclimber Thanks a lot for your contribution! You did great job. And all your questions are valid concerns. I will answer to them here.

> Should classes/types be annotated like packages, like functions and operators or in some other way?
> Same about modules:

I think all of them should be formatted like functions and operators. The rule is:

* `package` for packages
* _`partOfThePackage`_ for some part of some package (or other thing that belongs to some package)

> When reference in text can be resolved to module or to class/type, for example, Identity module and Identity type, what variant should be chosen?

I think it's better to chose specific type, not whole module. Modules can be big and contain irrelevant information (like different helper functions).

> Didn't find MonadIO and CoroutineT from Lecture 9, Stream comonad from Lecture 13.

`MonadIO` is here:

* https://hackage.haskell.org/package/base-4.10.1.0/docs/Control-Monad-IO-Class.html#t:MonadIO

Use new Hoogle interface to find it:

* http://hoogle.haskell.org/?hoogle=MonadIO&scope=set%3Astackage

`CoroutineT` from slides is not in any library actually. It's on wiki: 

* https://en.wikibooks.org/wiki/Haskell/Continuation_passing_style#Example:_coroutines

I think in that case you can just add link to this type, no special formatting with ` or similar is required.

`Stream` doesn't exist, so no formatting at all is required for it.

> Unclear how to format Lecture 16: Idris and Uncovered Topics (left them as is).

It's okay to not touch `Lecture 16: Idris`. `Idris` will be replaced by `Agda` this year (I hope) and I will format it manually. Regarding _uncovered topics_: same suggestions should be applied. Try to find everything you can. Some tips:

* `Generics` is `GHC.Generics` module
* Type families is extension
* Pragmas should be formatted as links to docs in GHC User Guide
* LiquidHaskell is package
* Arrows is both extension and `Control.Arrow` module in `base` package
* Link to purescript main website
* Same for `Agda`
* Zygo is this: https://wiki.haskell.org/Zygohistomorphic_prepromorphisms

File: README.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ Templates for homework: https://github.com/ChShersh/fp-homework-templates
   + **let** (variable declaration)
   + **where** clause
   + **if** expression
-  + **guards**
+  + Guards
   + **case** expression
   + Higher order functions
   + Lambdas (anonymous functions)
```
Comment (chshersh) [inline README.md:104]: It's better to add to course of this plan which extensions were used here: `-XTupleSections`, `-XLambdaCase` and `-XViewPatterns`. Also: `($)` and `(.)` should be formatted properly.

File: README.md [hunk 2]
```diff
@@ -122,73 +122,73 @@ Templates for homework: https://github.com/ChShersh/fp-homework-templates
...
++ [_`Bifunctor`_](http://hackage.haskell.org/package/base/docs/Data-Bifunctor.html#t:Bifunctor)
++ [_`Applicative`_](http://hackage.haskell.org/package/base/docs/Control-Applicative.html#t:Applicative)
+  * [_`ZipList`_](http://hackage.haskell.org/package/base/docs/Control-Applicative.html#t:ZipList)
++ [_`liftAN`_](http://hackage.haskell.org/package/base/docs/Control-Applicative.html#v:liftA) & Applicative style programming
++ [_`Alternative`_](http://hackage.haskell.org/package/base/docs/Control-Applicative.html#t:Alternative)
 + List comprehension syntax sugar
-+ _Traversable_ type class (and instances for _Maybe_, _List_)
++ [_`Traversable`_](http://hackage.haskell.org/package/base/docs/Data-Traversable.html#t:Traversable) type class (and instances for [_`Maybe`_](http://hackage.haskell.org/package/base/docs/Prelude.html#t:Maybe), [_`List`_](http://hackage.haskell.org/package/mtl/docs/Control-Monad-List.html#t:ListT))
 + Automatic deriving
   * [DeriveFunctor](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-functor-instances)
```
Comment (chshersh) [inline README.md]: This should _-XDeriveFunctor_

File: README.md [hunk 3]
```diff
@@ -122,73 +122,73 @@ Templates for homework: https://github.com/ChShersh/fp-homework-templates
...
   * [DeriveFoldable](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-foldable-instances)
```
Comment (chshersh) [inline README.md]: This should be _-XDeriveFoldable_ and _-XDeriveTraversable_

File: README.md [hunk 4]
```diff
@@ -203,110 +203,111 @@ Templates for homework: https://github.com/ChShersh/fp-homework-templates
...
 #### Presentation: http://slides.com/fp-ctd/lecture-7#/
 
 ## Lecture 8: RealWorld (by [Arseniy Seroka](https://github.com/jagajaga "jagajaga's GitHub")) [↑](https://github.com/jagajaga/FP-Course-ITMO#course-plan)
 + Building _IO_ system from scratch
-+ Introduce _IO_ monad
-+ __do__ notation
++ Introduce [_`IO`_](http://hackage.haskell.org/package/base/docs/Prelude.html#t:IO) monad
++ **do** notation
   * Syntax sugar
   * Applicative Do
```
Comment (chshersh) [inline README.md]: This is extension: _-XApplicativeDo_

File: README.md [hunk 5]
```diff
@@ -203,110 +203,111 @@ Templates for homework: https://github.com/ChShersh/fp-homework-templates
...
   * RebindableSyntax
```
Comment (chshersh) [inline README.md]: This is extension: _-XRebindableSyntax_

**Review by chshersh — CHANGES_REQUESTED** [review]

**Review by chshersh — APPROVED** [review]
Looks much better now! Thanks a lot again :+1:

**Comment by zclimber** [comment]
Corrections done.
Also, the monad I didn't find turned out to be MaybeIO, for which I decided to use https://wiki.haskell.org/Monad_Transformers_Tutorial link, and not MonadIO (((

**Pull request merged**